src/main/java/slave/elements/Storage.java

```diff
@@ -0,0 +1,173 @@
+package slave.elements;
+
+import slave.task.Task;
+import slave.task.Deadline;
+import slave.task.Event;
+import slave.task.ToDo;
+
+import slave.exception.DukeException;
+import slave.exception.InOutWentWrongException;
+import slave.exception.NoStorageFileDetectedException;
+import slave.exception.UnableToReadFileException;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.File;
+import java.io.PrintWriter;
+import java.io.FileWriter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import java.nio.file.Files;
+
+/**
+ * Storage class that writes and loads data from local .txt file.
+ */
+public class Storage {
+
+    private File file;
+
+    /**
+     * Constructor that initialises a File object.
+     *
+     * @param filePath FilePath of .txt file to write/load.
+     */
+    public Storage(String filePath) {
+        this.file = new File(filePath);
+    }
+
+    /**
+     * Clears storage of data.
+     *
+     * @throws NoStorageFileDetectedException When storage file not found.
+     */
+    void clearStorage() throws NoStorageFileDetectedException {
+        try {
+            PrintWriter writer = new PrintWriter(this.file);
+            writer.close();
+        } catch (FileNotFoundException e) {
+            throw new NoStorageFileDetectedException();
+        }
+    }
+
+    /**
+     * Loads tasks from .txt file.
+     *
+     * @return An ArrayList of tasks parsed from .txt file.
+     * @throws DukeException Throws if file cannot be found or input went wrong.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> taskList;
+        try {
+            taskList = new ArrayList<>();
+            List<String> list = Files.readAllLines(this.file.toPath());
+            int index = 1;
+            for (String line : list) {
+                taskList.add(formatFileToTask(line, index));
+                index++;
+            }
+        } catch (IOException e) {
+            throw new InOutWentWrongException();
+        }
+        return taskList;
+    }
+
+    /**
+     * Refreshes storage and updates list of tasks to the stored local .txt file.
+     *
+     * @param taskList Current task list.
+     * @throws DukeException Throws if file cannot be found.
+     */
+    void refreshStorage(ArrayList<Task> taskList) throws DukeException {
+        try {
+            PrintWriter writer = new PrintWriter(this.file);
+            writer.close();
+            for (Task task : taskList) {
+                addTask(task);
+            }
+        } catch (FileNotFoundException e) {
+            throw new NoStorageFileDetectedException();
+        }
+    }
+
+    /**
+     * Encodes task and add it to .txt file.
+     *
+     * @param task Task to store in .txt file.
+     * @throws DukeException Throws if input went wrong in writing file.
+     */
+
+    void addTask(Task task) throws DukeException {
```

Comment:
I like how your methods for formatting the tasks in and out of the text file is so clean!